### PR TITLE
layers: lx464 Add drawtime validation of bound buffer memory.

### PR DESCRIPTION
**Pull request by MarkY-LunarG**
Add a drawtime check to validate that storage or uniform buffers
have valid memory bound to them at draw time before using.

Change-Id: Ia0a07f221dfcb2ea4315e4f52ee04d7cdda22cf7

### CODE REVIEW

**Comment by lentinem** [comment]
+1

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2736,42 +2736,78 @@ static bool validate_and_update_drawtime_descriptor_state(
...
+                                auto buffer_node = dev_data->bufferMap.find(pWDS->pBufferInfo[j].buffer);
+                                if (buffer_node == dev_data->bufferMap.end()) {
+                                    result |= log_msg(
+                                        dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT,
+                                        VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_SET_EXT,
+                                        reinterpret_cast<const uint64_t &>(set_node->set), __LINE__, DRAWSTATE_INVALID_BUFFER, "DS",
+                                        "VkDescriptorSet (%#" PRIxLEAST64 ") %s (%#" PRIxLEAST64 ") at index #%u"
+                                        " is not defined!  Has vkCreateBuffer been called?",
+                                        reinterpret_cast<const uint64_t &>(set_node->set),
+                                        string_VkDescriptorType(pWDS->descriptorType), pWDS->pBufferInfo[j].buffer, i);
```
Comment (tobine) [inline layers/core_validation.cpp]: buffer needs reinterpret_cast, I'll fix this

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2736,42 +2736,78 @@ static bool validate_and_update_drawtime_descriptor_state(
...
+                                    auto mem_entry = dev_data->memObjMap.find(buffer_node->second.mem);
+                                    if (mem_entry == dev_data->memObjMap.end()) {
+                                        result |= log_msg(
+                                            dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT,
+                                            VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_SET_EXT,
+                                            reinterpret_cast<const uint64_t &>(set_node->set), __LINE__, DRAWSTATE_INVALID_BUFFER,
+                                            "DS", "VkDescriptorSet (%#" PRIxLEAST64 ") %s (%#" PRIxLEAST64 ") at index"
+                                                  " #%u, has no memory bound to it!",
+                                            reinterpret_cast<const uint64_t &>(set_node->set),
+                                            string_VkDescriptorType(pWDS->descriptorType), pWDS->pBufferInfo[j].buffer, i);
```
Comment (tobine) [inline layers/core_validation.cpp]: buffer needs reinterpret_cast, I'll fix this

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**